packages/server-wallet/src/protocols/open-channel.ts

```diff
@@ -122,7 +122,7 @@ const signPostFundSetup = (ps: ProtocolState): ProtocolResult | false =>
   });
 
 const completeOpenChannel = (ps: ProtocolState): CompleteObjective | false =>
-  (isRunning(ps.app.supported) || isPostfundSetup(ps.app.supported)) &&
+  (isRunning(ps.app.supported) || ps.app.supported === ps.app.participants.length * 2 - 1) &&
```

Comment:
I'd like to note this as an argument for changing the nitro protocol logic.

Suppose we change the adjudicator to have the following rules:
A transition from state `s_n` to state `s_m` is valid if and only if:
R1. `n > 0` and `m == n+1` (these are turn numbers)
R2. `s_n` and `s_m` are not final
R3. `AppRules.validTransition(s_n, s_m)` is true
R4. `constantPart(s_n) == constantPart(s_m)`

In other words, turns 0 & 1 are specially reserved for pre- and post-FS states. This offers the same safety* as the current set of rules.

* I am 99% sure of this.

# Benefits
B1. This is a simpler set of rules than the current implementation. It is easier for us to audit, and easier to explain to other folks _conditional on being able to explain signatures_.
B2. This is more gas efficient in both the happy path (we force conclusion proofs to be single states) and the sad path (we force challenges before running states

# Claimed neutrals
CN3. Wallets can get the same benefits with the current implementation, since they can opt to sign states `0` and then states `3`.

# Drawbacks
D4. We would have to change the contracts.
D5. It seemed to be the consensus preference to take turns for pre- and post-FS.  We could not opt into turn-taking if we wanted to.

I am happy for others to add to this. I am biased.

# Claimed?
I say that (CN3) is a _claimed_ neutral because, as is evidenced here, the current nitro rules allow for more foot guns. The following are three foot guns that are avoided by the rules R1-R4.

## Footgun 1
Suppose I want to know "Is this channel running? Is it safe for me to try to make an update?"
### Current answer
A. BOTH `channel.supportedState.turnNumber >= channel.participants.length * 2 - 1`
B. AND "I have not seen any `final` states"

The error in `master` is that (A) is replaced with `isPostFundSetup(channel.supported) || isRunning(channel.supported)`, which is succinct, but wrong.

### Simpler answer
A. BOTH `channel.supportedState.turnNumber >= 1`
B. AND "I have not seen any `final` states"

In this case, (A) is now equivalent to `isPostFundSetup(channel.supported) || isRunning(channel.supported)`.
So, the developer who wrote this line's mental model is in line with the correct answer (in spite of it being more complicated.)

It is a good property for mental models to be correct. (It is a better property for them to be correct _and_ simple!)

## Footgun 2
Suppose I want to quickly write a wallet.
Then I will assume that each channel has two participants, and use the (often magic) constant `3` instead of the correct, dynamic number `channel.participants.length * 2 - 1`.

You might not have noticed my accidental use of this magic constant in (CN3). I didn't when I wrote it.

## Footgun 3
Suppose I want to quickly write a wallet.
Then I might assume that the outcome of the channel has one outcome per participant, _in the same order as the participants_
This would lead me to make assumptions about when a participant should sign a post fund state based on its _index in the channel_

# Drawbacks?
D4: We are going to change the contracts anyway, post-audit.
  - This retort has another drawback, that the audit would now be a bit stale?
D5. I assert that the consensus was wrong, and that turn-taking while funding nudges our mental models to be slightly wrong, manifesting in bugs like this one.

Reply:
Seems reasonable to me.